src/hotspot/share/opto/superword.cpp

```diff
@@ -38,146 +38,51 @@
-  _bb_idx(arena(), (int)(1.10 * phase->C->unique()), 0, 0), // node idx to index in bb
-  _block(arena(), 8,  0, nullptr),                          // nodes in current block
-  _data_entry(arena(), 8,  0, nullptr),                     // nodes with all inputs from outside
-  _mem_slice_head(arena(), 8,  0, nullptr),                 // memory slice heads
-  _mem_slice_tail(arena(), 8,  0, nullptr),                 // memory slice tails
-  _node_info(arena(), 8,  0, SWNodeInfo::initial),          // info needed per node
-  _clone_map(phase->C->clone_map()),                        // map of nodes created in cloning
+SuperWord::SuperWord(Arena* arena, const VLoop &vloop) :
+  _vloop(vloop),
+  _arena(arena),
+  _packset(arena, 8,  0, nullptr),                          // packs for the current block
+  _bb_idx(arena, vloop.estimated_node_count(), 0, 0),       // node idx to index in bb
+  _block(arena, vloop.estimated_body_length(), 0, nullptr), // nodes in current block
+  _data_entry(arena, 8,  0, nullptr),                       // nodes with all inputs from outside
+  _mem_slice_head(arena, 8,  0, nullptr),                   // memory slice heads
+  _mem_slice_tail(arena, 8,  0, nullptr),                   // memory slice tails
+  _node_info(arena, vloop.estimated_body_length(), 0, SWNodeInfo::initial), // info needed per node
+  _clone_map(phase()->C->clone_map()),                      // map of nodes created in cloning
   _align_to_ref(nullptr),                                   // memory reference to align vectors to
-  _dg(_arena),                                              // dependence graph
-  _nlist(arena(), 8, 0, nullptr),                           // scratch list of nodes
-  _lpt(nullptr),                                            // loop tree node
-  _lp(nullptr),                                             // CountedLoopNode
-  _loop_reductions(arena()),                                // reduction nodes in the current loop
-  _bb(nullptr),                                             // basic block
-  _iv(nullptr),                                             // induction var
+  _dg(arena),                                               // dependence graph
+  _nlist(arena, vloop.estimated_body_length(), 0, nullptr), // scratch list of nodes
+  _loop_reductions(arena),                                  // reduction nodes in the current loop
   _race_possible(false),                                    // cases where SDMU is true
-  _early_return(true),                                      // analysis evaluations routine
-  _do_vector_loop(phase->C->do_vector_loop()),              // whether to do vectorization/simd style
+  _do_vector_loop(phase()->C->do_vector_loop()),            // whether to do vectorization/simd style
   _num_work_vecs(0),                                        // amount of vector work we have
   _num_reductions(0)                                        // amount of reduction work we have
```

Comment:
For some data structures I now pre-allocate memory for the maximum size they may ever reach. This is to avoid re-allocation when they grow.
Before this change, we used to only create SuperWord once, and use it for all loops in the compilation. Now that I ripped out the "init" method, and avoid reusing SuperWord this way, we want to make sure we do not re-allocate too much.